### ISSUE TITLE
Error creating extension

### ISSUE THREAD

**Issue by tharris9d** [valadas/Eraware_Dnn_Templates#435]
Eraware_Dnn_Templates v1.10

VS Community 2022 Run as Administrator
Module attempted creation in DesktopModules folder
run -> debug package

All .net installed
Microsoft.AspNetCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
PS C:\Users\Tom> dotnet --list-sdks
3.1.416 [C:\Program Files\dotnet\sdk]
5.0.404 [C:\Program Files\dotnet\sdk]
6.0.101 [C:\Program Files\dotnet\sdk]

Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()

███╗   ██╗██╗   ██╗██╗  ██╗███████╗
████╗  ██║██║   ██║██║ ██╔╝██╔════╝
██╔██╗ ██║██║   ██║█████╔╝ █████╗
██║╚██╗██║██║   ██║██╔═██╗ ██╔══╝
██║ ╚████║╚██████╔╝██║  ██╗███████╗
╚═╝  ╚═══╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

NUKE Execution Engine version 5.3.0 (Windows,.NETCoreApp,Version=v5.0)

The SDK 'Microsoft.NET.Sdk' specified could not be found.  E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\_build\_build.csproj
StackTrace:
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult, Boolean throwOnFileNotExistsError)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement, SdkResult& sdkResult)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project..ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.FromProjectRootElement(ProjectRootElement rootElement, ProjectOptions options)
   at Nuke.Common.ProjectModel.ProjectModelTasks.ParseProject(String projectFile, String configuration, String targetFramework)
   at Nuke.Common.Execution.Telemetry.CheckAwareness()
Assertion failed: Could not find git directory for 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa'.
StackTrace:
   at Nuke.Common.ControlFlow.Fail(String text, Exception exception)
   at Nuke.Common.Git.GitRepository.FromLocalDirectory(String directory)
   at Nuke.Common.Git.GitRepositoryAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Assertion failed: Could not find git directory for 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa'.
> "C:\Program Files\dotnet\dotnet.exe" C:\Users\Tom\.nuget\packages\gitversion.tool\5.3.7\tools\netcoreapp3.1\any\gitversion.dll /nocache /nofetch
Process 'dotnet.exe' exited with code 1.
   > "C:\Program Files\dotnet\dotnet.exe" C:\Users\Tom\.nuget\packages\gitversion.tool\5.3.7\tools\netcoreapp3.1\any\gitversion.dll /nocache /nofetch
   @ E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\_build\bin\Debug
Error output:
   Cannot find the .git directory

StackTrace:
   at Nuke.Common.Tooling.ProcessExtensions.AssertZeroExitCode(IProcess process)
   at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(GitVersionSettings toolSettings)
   at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(Configure`1 configurator)
   at Nuke.Common.Tools.GitVersion.GitVersionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()

╬════════════════════
║ CleanDocsFolder
╬═══════════

Cleaning directory 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\docs'...

╬════════════════════════
║ SetManifestVersions
╬═══════════════

Found package Di_MyStencilSpa with version 00.01.00
Updated package Di_MyStencilSpa to version 00.01.00
Saved E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\manifest.dnn

╬══════════
║ Clean
╬══

Cleaning directory 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\artifacts'...
Cleaning directory 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\TestResults'...
Creating directory 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\TestResults\UnitTests'...
Creating directory 'E:\websites\DNN9102\DesktopModules\MyStencilSpa\MyStencilSpa\TestResults\IntegrationTests'...

╬════════════
║ Restore
╬═══

**Comment by valadas** [comment]
Thanks for the very informative bug report.

Can you try to edit the global.json in the root of the project with the following:

```
{
  "sdk": {
    "version": "5.0.302",
    "rollForward": "latestMajor"
  }
}
```

Let me know if that works...

**Comment by tharris9d** [comment]
I updated the global.json file and got slightly different errors.
Running as Administrator.

Creating directory 'E:\websites\DNN991\DesktopModules\MyStencilSpaModule1\MyStencilSpaModule1\.nuke\temp'...
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Creating directory 'E:\websites\DNN991\DesktopModules\MyStencilSpaModule1\MyStencilSpaModule1\.github\workflows'...
Creating directory 'E:\websites\DNN991\DesktopModules\MyStencilSpaModule1\MyStencilSpaModule1\.github\workflows'...
Creating directory 'E:\websites\DNN991\DesktopModules\MyStencilSpaModule1\MyStencilSpaModule1\.github\workflows'...
Exception has been thrown by the target of an invocation.
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Exception has been thrown by the target of an invocation.
StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nuke.Common.ProjectModel.SolutionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props
StackTrace:
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult, Boolean throwOnFileNotExistsError)
   at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement, SdkResult& sdkResult)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project..ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.FromProjectRootElement(ProjectRootElement rootElement, ProjectOptions options)
   at Nuke.Common.ProjectModel.ProjectModelTasks.ParseProject(String projectFile, String configuration, String targetFramework)
   at Nuke.Common.Execution.Telemetry.CheckAwareness()
GitHubActions (Build) configuration files have changed.
GitHubActions (PR_Validation) configuration files have changed.
GitHubActions (Release) configuration files have changed.
Press any key to continue...
ReflectionTypeLoadException: Unable to load one or more of the requested types.
Could not load file or assembly 'NuGet.Common, Version=6.0.0.278, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.ToArray()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.GetEnumerator()+MoveNext()
   at System.Linq.Enumerable.SelectIPartitionIterator`2.MoveNext()
   at Newtonsoft.Json.Linq.JContainer.AddInternal(Int32 index, Object content, Boolean skipParentCheck)
   at Nuke.Common.Execution.SchemaUtility.GetBuildSchema(NukeBuild build)
   at Nuke.Common.Execution.SchemaUtility.WriteBuildSchemaFile(NukeBuild build)
   at Nuke.Common.Execution.HandleShellCompletionAttribute.OnBuildCreated(NukeBuild build, IReadOnlyCollection`1 executableTargets)
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at Nuke.Common.NukeBuild.ExecuteExtension[TExtension](Expression`1 action)
   at Nuke.Common.Execution.BuildManager.Execute[T](Expression`1[] defaultTargetExpressions)

**Comment by valadas** [comment]
Hmm, that is odd. I am still thinking maybe some sort of permission issues, can you right-click the project folder and temporarily set it to everyone just to rule this out (remove that permission after the test to be safe).

Also, if you open a PowerShell (as admin too) in that folder and run `.\build.ps1` do you get the same error or it works or have different errors?

Sorry for the wild guesses, just trying to figure out what could possible cause this. At first glance it is not finding Nuget, but it is a declared dependency in the build.csproj file...

Two other things pop to mind
1. Is that computer where the build runs connected to the internet ?
2. If you have an antivirus, can you pause it just before running the build...

**Comment by tharris9d** [comment]
I tried turning off the Anti-virus, Firewall, gave full access to Everyone on the project folder and tried running build.ps1 with elevated permissions. I got the same errors. I have 1gb internet.

**Comment by valadas** [comment]
Ok, thanks, I am not sure what could be going on. Currently trying to upgrade to Nuke 6 but running into some different issues...

**Comment by valadas** [comment]
Hey @tharris9d thanks a lot for the contribution, I am sorry I was not able to help much with this issue as it does not happen to me.

But something poped to mind, when you got the first dialog before the project wizard, was the checkbox "place project and solution in the same folder" checked or unchecked ? Either way if you can try the opposite with a new project name to test that out...

**Comment by tharris9d** [comment]
Will do. 

⁣Get BlueMail for Android ​

On Jan 13, 2022, 10:39 PM, at 10:39 PM, Daniel Valadas ***@***.***> wrote:
>Hey @tharris9d thanks a lot for the contribution, I am sorry I was not
>able to help much with this issue as it does not happen to me.
>
>But something poped to mind, when you got the first dialog before the
>project wizard, was the checkbox "place project and solution in the
>same folder" checked or unchecked ? Either way if you can try the
>opposite with a new project name to test that out...
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/valadas/Eraware_Dnn_Templates/issues/435#issuecomment-1012814015
>You are receiving this because you were mentioned.
>
>Message ID:
***@***.***>

**Comment by tharris9d** [comment]
Sir Daniel, 
I have made some progress. 
1. I had to turn MSBuildEnableWorkloadResolver off by adding to the Environment Variables with a 0
2. I added Git.exe and the cmd folder to the Environment Path.
3. I checked the Place Project in the same folder as Solution

Now I am just getting the following NUKE error

███╗   ██╗██╗   ██╗██╗  ██╗███████╗
████╗  ██║██║   ██║██║ ██╔╝██╔════╝
██╔██╗ ██║██║   ██║█████╔╝ █████╗
██║╚██╗██║██║   ██║██╔═██╗ ██╔══╝
██║ ╚████║╚██████╔╝██║  ██╗███████╗
╚═╝  ╚═══╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝

NUKE Execution Engine version 5.3.0 (Windows,.NETCoreApp,Version=v5.0)

Assertion failed: Could not find git directory for 'E:\websites\DNN991\DesktopModules\MyModule3'.
StackTrace:
   at Nuke.Common.ControlFlow.Fail(String text, Exception exception)
   at Nuke.Common.Git.GitRepository.FromLocalDirectory(String directory)
   at Nuke.Common.Git.GitRepositoryAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()
Assertion failed: Could not find git directory for 'E:\websites\DNN991\DesktopModules\MyModule3'.
> "C:\Program Files\dotnet\dotnet.exe" C:\Users\Tom\.nuget\packages\gitversion.tool\5.3.7\tools\netcoreapp3.1\any\gitversion.dll /nocache /nofetch
Process 'dotnet.exe' exited with code 1.
   > "C:\Program Files\dotnet\dotnet.exe" C:\Users\Tom\.nuget\packages\gitversion.tool\5.3.7\tools\netcoreapp3.1\any\gitversion.dll /nocache /nofetch
   @ E:\websites\DNN991\DesktopModules\MyModule3\_build\bin\Debug
Error output:
   Cannot find the .git directory

StackTrace:
   at Nuke.Common.Tooling.ProcessExtensions.AssertZeroExitCode(IProcess process)
   at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(GitVersionSettings toolSettings)
   at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(Configure`1 configurator)
   at Nuke.Common.Tools.GitVersion.GitVersionAttribute.GetValue(MemberInfo member, Object instance)
   at Nuke.Common.ValueInjection.ValueInjectionAttributeBase.<>c__DisplayClass0_0.<TryGetValue>b__0()

╬════════════════════
║ CleanDocsFolder
╬═══════════

Cleaning directory 'E:\websites\DNN991\DesktopModules\MyModule3\docs'...

╬════════════════════════
║ SetManifestVersions
╬═══════════════

Found package Di_MyModule3 with version 00.01.00
Updated package Di_MyModule3 to version 00.01.00
Saved E:\websites\DNN991\DesktopModules\MyModule3\manifest.dnn

╬══════════
║ Clean
╬══

Cleaning directory 'E:\websites\DNN991\DesktopModules\MyModule3\artifacts'...
Cleaning directory 'E:\websites\DNN991\DesktopModules\MyModule3\TestResults'...
Creating directory 'E:\websites\DNN991\DesktopModules\MyModule3\TestResults\UnitTests'...
Creating directory 'E:\websites\DNN991\DesktopModules\MyModule3\TestResults\IntegrationTests'...

╬════════════
║ Restore
╬═══

> "C:\Program Files\dotnet\dotnet.exe" restore E:\websites\DNN991\DesktopModules\MyModule3\Module.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
> "C:\Program Files\dotnet\dotnet.exe" restore E:\websites\DNN991\DesktopModules\MyModule3\UnitTests\UnitTests.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
> "C:\Program Files\dotnet\dotnet.exe" restore E:\websites\DNN991\DesktopModules\MyModule3\IntegrationTests\IntegrationTests.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.

╬════════════
║ Compile
╬═══


E:\websites\DNN991\DesktopModules\MyModule3\_build\bin\Debug\_build.exe (process 10276) exited with code -1.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . . 

![image](https://user-images.githubusercontent.com/217445/149579101-43497f37-2ba7-41ad-a832-5c3d616a65ee.png)

**Comment by valadas** [comment]
Oh, great! nice progress there, soo for the .git directory part, this should be a warning in theory and still let the build run (as it does continue with next targets I see).

After the compile header there was no other details showing up other than existed with code -1 ?

**Comment by tharris9d** [comment]
This is what happens....
![image](https://user-images.githubusercontent.com/217445/149587755-ba33253a-cabf-4512-8087-f4aa4a106973.png)

**Comment by david-poindexter** [comment]
Perhaps this is a MSBuild (build tools) issue.  See https://docs.microsoft.com/en-us/visualstudio/msbuild/walkthrough-using-msbuild?view=vs-2022 for more help to ensure it is good.

**Comment by tharris9d** [comment]
@david-poindexter Thanks, I am working to track down the msbuild issues.

**Comment by tharris9d** [comment]
Progress:
I hard-coded the path to MSBUILD
Target Compile, Target UnitTests and Target IntegrationTests
MSBuildTasks.MSBuild(s => s
     .SetProcessToolPath(@"D:\Program Files\Microsoft Visual Studio\2022\Community\MSbuild\Current\Bin\msbuild.exe")
It worked with some errors. @valadas Is there an Install package created or is that not in this template.

**Comment by david-poindexter** [comment]
@tharris9d yes, there is an install package created if you change to the `Package` build target.

**Comment by tharris9d** [comment]
Here is the entire result. Release Package
https://gist.github.com/tharris9d/36d98730de23738423d7f99ab468d0fb

**Comment by david-poindexter** [comment]
Thanks great @tharris9d

**Comment by valadas** [comment]
Excellent, thanks for the help on this @david-poindexter 

So @tharris9d the package will output an install zip to the Artifacts folder of the project, it will also copy this file to the DNN install/module folder so if you open the Persona Bar and go to Available Extensions of that local site, the module should show as available.

Because there was some partial builds and bugs along the way, you probably no longer see the HTML startup document in Visual Studio but from this point you should be able to follow these instructions : https://github.com/valadas/Eraware_Dnn_Templates/blob/develop/Eraware_Dnn_Spa_Ef_Di_Stencil/module/docfx_project/articles/git.md

Now, you should not need to hardcode that path, there might be some issue with how Nuke locates that, but my next version will upgrade to Nuke 6 (finally getting to the point of ironing out the bugs I was having with it). And the next template will also declare the VisualStudio dependencies automatically, so it may fix that locating of it issue too... I'll leave this open to contact you with that next beta.

**Comment by tharris9d** [comment]
Thanks @valadas and @david-poindexter

**Comment by david-poindexter** [comment]
You are most welcome - so glad you find the template useful.  It is definitely our "go-to" for DNN module development now!

**Comment by tharris9d** [comment]
Very cool that the module writes to the install folder / modules!!! I'm excited to work with web components!!!

**Comment by valadas** [comment]
And after, that you can your `deploy` to deploy to the test site for backend/frontend updates, and to work quick you can use `watch` to simply watch for frontend file changes and just reload the browser to see the new frontend without a full deploy... Almost live-reload...

**Comment by tharris9d** [comment]
@valadas NICE!!!

**Comment by valadas** [comment]
A big thanks to @matkoch too on helping with out upgrading Nuke here for the next release :)

**Comment by tharris9d** [comment]
Thanks @matkoch

**Comment by tharris9d** [comment]
@david-poindexter @valadas I would like to add the nvQuickComponents to my new Stencil module. Are there any tutorial docs? I am sure I can figure it out by following the pattern but it would be nice to read a doc if one is available. Thanks in advance. 
FYI I added in the changes to the VSIX project (global.json and hardcoded path to MSBUILD), reinstalled it in VS2022 and it's working great. I installed the module and it is working!!!!

**Comment by valadas** [comment]
Oh, you should be able to simply `npm install @nvisionative/nvq-components` in the module.web folder and be ready to use them. Then in your own components if you want to use them, you go by component like `<nvq-progress-bar />` or some such and you can check the auto-generated docs of each component directly on github in the folder of each component to know what they support... like https://github.com/nvisionative/nvQuickComponents/tree/develop/src/components/nvq-progress-bar

**Comment by david-poindexter** [comment]
Yes, exactly what he said ^^^  😆 

@tharris9d by the way, not sure if you are aware of the `dnn-elements` project, but we (mostly @valadas) have made great progress on it.  It will be moving to a DNN Community org repo soon, but it is currently at https://github.com/eraware/dnn-elements.  Check out the [demo site](https://eraware.github.io/dnn-elements/) too!

**Comment by tharris9d** [comment]
The dnn-elements are amazing! Have you worked with any web component datatable/datagrids?

**Comment by david-poindexter** [comment]
@tharris9d check out https://github.com/revolist/revogrid - I played around it a bit a while back and it was pretty amazing.

**Comment by valadas** [comment]
Ok @tharris9d I just released v1.2.0 with a few notable updates which uses Nuke6.0.1 which has some fixes to finding MsBuild paths (so you should in theory no longer need to hardcode the path). Note that Nuke now follows .Net versioning so Nuke 6 requires .Net6, if you get into .Net issues please install the .Net 6 SDK using Visual Studio Installer or here https://dotnet.microsoft.com/en-us/download/dotnet/6.0

For the new release, you should see that in Visual Studio Extenions under updates or you can grab it from here https://marketplace.visualstudio.com/items?itemName=DanielValadas.eraware-dnn-templates or here https://github.com/valadas/Eraware_Dnn_Templates/releases/tag/v1.2.0

**Comment by valadas** [comment]
I gonna go ahead and close this issue, but if you run into other problems with this new version, feel free to open a new issues describing that.
